src/headers/tomcrypt_pk.h

```diff
@@ -45,6 +45,7 @@ typedef struct Rsa_key {
 } rsa_key;
 
 int rsa_make_key(prng_state *prng, int wprng, int size, long e, rsa_key *key);
+int rsa_make_key_bn_e(prng_state *prng, int wprng, int size, void *e, rsa_key *key);
```

Comment:
I like the idea, but we don't accept API's anymore which have MPI-specific arguments.

In the spirit of the `rsa_set_xyz()` functions could you please change it to:

```suggestion
int rsa_make_key_bn_e(prng_state *prng, int wprng, int size, const unsigned char *e, unsigned long elen, rsa_key *key);
```

I'm also not too much a fan of the name `rsa_make_key_bn_e()` but I don't have a better idea right now. Maybe you @karel-m ?

I didn't review the implementation yet, will do so after you changed the API.

Reply:
@sjaeckel 
Sure I can change the function prototype as you suggested.
Concerning the function name, I'm not fan with it too but it was the most explicit one I found. Otherwise I can propose;
- `rsa_make_key2`
- `rsa_make_key_ex`

I'll add a test and check documentation. I just wanted to be sure first that idea was accepted. 
Do you want separated commit for the test?

Reply:
> Concerning the function name, I'm not fan with it too but it was the most explicit one I found.

yes, you've already chosen the best of those three proposals :)

> Do you want separated commit for the test?

I don't really care :)

It's nice if commits are separate and self-contained, but as long as it's all there it's fine.

Reply:
I would choose: `int rsa_make_key_hex_e(prng_state *prng, int wprng, int size, const char *hex_e, rsa_key *key);`

Reply:
I kind of like the idea, but I don't like the difference in API to the `rsa_set_*()` functions ... also I'm not sure whether that's really such a good idea as hex is maybe a good representation for "human interaction" but I'm pretty sure that "on the wire" this is going to be something else than a hex string ... even though in some use cases it's going to be `base16_decode() && rsa_make_key_bn_e()` ...

I've just looked into the use case of @cneveux and the op-tee integration would indeed be easiest with the initial approach where `e` is a "mpi `void*`", so I think we should take the more efficient variant, both runtime and memory wise and that's `int rsa_make_key_bn_e(prng_state *prng, int wprng, int size, void *e, rsa_key *key);`

Fine @karel-m ?

Reply:
As op-tee is statically linking against ltc we could also provide a `int rsa_make_key_bn_e(prng_state *prng, int wprng, int size, void *e, rsa_key *key);` in `tomcrypt_private.h` so you don't have to convert back and forth.

@cneveux @jforissier @karel-m what do you think?

Reply:
@sjaeckel sounds good to me

Reply:
Hmm, well, at least leave a comment in `tomcrypt_private.h` like `/* used by op-tee */`

Reply:
ok,
* so I'll add and document this API, plus add a test.
`int rsa_make_key_hex_e(prng_state *prng, int wprng, int size, const char *hex_e, rsa_key *key);`

* and add but no document this API, no test needed too, as `rsa_make_key_hex_e()` and this `rsa_make_key_bn_e()` will call the same static function.  
`int rsa_make_key_bn_e(prng_state *prng, int wprng, int size, void *e, rsa_key *key); /* used by op-tee */` 

Do you agree?